src/client/qwaylandextendedsurface.cpp

```diff
@@ -91,7 +91,14 @@ void QWaylandExtendedSurface::setContentOrientationMask(Qt::ScreenOrientations m
 
 void QWaylandExtendedSurface::extended_surface_onscreen_visibility(int32_t visibility)
 {
-    m_window->window()->setVisibility(static_cast<QWindow::Visibility>(visibility));
+    // If visibility is Hidden we want the window to just stop rendering, not to hide.
+    // calling setVisible(false) causes a NULL buffer to be attached, and it is better
+    // to show an outdated buffer rather than random garbage from memory.
+    QRegion region;
+    if (visibility != QWindow::Hidden) {
+        region = QRegion(QRect(0, 0, m_window->window()->width(), m_window->window()->height()));
+    }
+    QWindowSystemInterface::handleExposeEvent(m_window->window(), region);
```

Comment:
This will hide the entire root window though.. Is that what you intended? Sending an obscure event like you do here, should result in that the scene graph stops rendering all together.